apps/src/templates/instructions/InstructionsWithWorkspace.jsx

```diff
@@ -12,43 +12,40 @@ var instructions = require('../../redux/instructions');
  * Owns maxHeightAvailable for instructions, updating as appropriate on window
  * resize events
  */
-var InstructionsWithWorkspace = React.createClass({
-  propTypes: {
+export class UnwrappedInstructionsWithWorkspace extends React.Component {
```

Comment:
The default export for this file (which is unchanged by this PR) is at the end of the file, where we use [`react-redux`'s `connect` method](https://github.com/reduxjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options) to generate a new, "wrapped" version of the component that is the preferred way to use it.  (See [Radium](https://github.com/FormidableLabs/radium) for another example of wrapped/higher-order components.)

https://github.com/code-dot-org/code-dot-org/blob/6cb9db4e2dffb7853188683258c1f488c47e0c2e/apps/src/templates/instructions/InstructionsWithWorkspace.jsx#L105-L115

Changing the name of the unwrapped component doesn't require us to update any of the places that import this file because default exports are unnamed - the importing file doesn't know or care what the export was named, behind the scenes it's just `'default'` and can be renamed on import.

```js
// InstructionsWithWorkspace.jsx
export default class InstructionsWithWorkspace {...}

// Consumer.jsx
import FruitSalad from './InstructionsWithWorkspace'; // works fine!
```

You can read more about the difference between default and named exports [on the MDN page for the `export` keyword](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export#Description).

Testing Redux-connected components can require quite a bit of setup, though, so here I'm making the unwrapped version of the component a second, non-default export for unit testing purposes.  The unit tests import and rename the unwrapped component:

https://github.com/code-dot-org/code-dot-org/blob/6cb9db4e2dffb7853188683258c1f488c47e0c2e/apps/test/unit/templates/instructions/InstructionsWithWorkspaceTest.jsx#L6-L8

I could actually get away with _not_ adding "Unwrapped" to the name, but it can lead to some confusion because there's a subtle distinction in how you import it:

```js
// InstructionsWithWorkspace.jsx
export class InstructionsWithWorkspace {...}
export default connect(...)(InstructionsWithWorkspace);

// Consumer.jsx
// import wrapped version
import InstructionsWithWorkspace from './InstructionsWithWorkspace';
// import unwrapped version
import {InstructionsWithWorkspace} from './InstructionsWithWorkspace';
```

...so I prefer to add something to the name making it obviously the unwrapped version.

Reply:
Awesome! Thanks for the info!